### ISSUE TITLE
Label text not updated after calling open_settings()

### ISSUE THREAD

**Issue by avonderlieth** [kivy/kivy#8519]
**Software Versions**
* Python: 3.9.2
* OS: WIN32 and Raspberry Pi (bullseye)
* Kivy: 2.2.1
* Kivy installation method: pip install kivy

Description:
After calling open_settings(), label texts are no longer updated on the screen. In the background the values and text properties do change and the color property will be updated, but the GUI still shows the text as it was before the first call to open_settings()

This gist shows the effect using the pong example expanded by a settings object and the call to open_settings() whenever player2 scores. All is well until the first time open_settings() is called, i.e. the label texts and colors are being updated after a score. However, after the first open_settings, only the label colors will be updated:
[https://gist.github.com/avonderlieth/12669a4b4ff68da8a09662c29927763f ](https://gist.github.com/avonderlieth/12669a4b4ff68da8a09662c29927763f)

This is just an example to illustrate the behavior. My original project (much larger) was working with Kivy-1.9.1 but after a (forced) upgrade to 2.2.1 the code was no longer working. The behavior is present in both WIN32 and Raspian (bullseye)

**Expected behavior**
label text and label color are being updated after a score

**To Reproduce**
See example in gist

**Comment by ElliotGarbus** [comment]
I was able to replicate the issue on Windows 11, using the default kivy settings screen.  Here is a minimal reproducer.  While the F1 key can be pressed to open the settings panel, I added a button to as well.
The clock continues to run, but the screen does not update.  Note that setting the text attribute of the Label in the callback shows the same behavior.  

```python
from kivy.app import App
from kivy.lang import Builder
from kivy.clock import Clock
from kivy.properties import NumericProperty

kv = """
BoxLayout:
    orientation: 'vertical'
    Label:
        id: label
        text: str(app.count)
        font_size: 60
    Button:
        size_hint_y: None
        height: dp(48)
        text: 'Open Settings'
        on_release: app.open_settings()
"""


class CheckSettings(App):
    count = NumericProperty(0)

    def build(self):
        return Builder.load_string(kv)

    def on_start(self):
        Clock.schedule_interval(self._count, 1)

    def _count(self, _):
        self.count += 1
        print(f'Count is now {self.count}')


CheckSettings().run()
```

**Comment by ElliotGarbus** [comment]
Interesting the bug does not appear if kivy.uix.settings is imported prior to opening the settings panel:
@avonderlieth this is a potential workaround to your issue.

I've extended the example to put the settings panel beside the Label so I could better see when the change was occurring.  

```python
from kivy.app import App
from kivy.lang import Builder
from kivy.clock import Clock
from kivy.properties import NumericProperty
from kivy.uix.popup import Popup
from kivy.uix.label import Label
import kivy.uix.settings  # BUG does not occur if this line is here!

kv = """
BoxLayout:
    BoxLayout:
        orientation: 'vertical'
        Label:
            id: label
            text: str(app.count)
            font_size: 60
        BoxLayout:
            size_hint_y: None
            height: dp(48)
            Button:
                text: 'Popup'
                on_release: app.pop()
            Button:
                text: 'Open Settings'
                on_release: app.open()
    BoxLayout:
        id: box
"""


class CheckSettings(App):
    count = NumericProperty(0)

    def __init__(self, **kwargs):
        super().__init__(**kwargs)
        self.settings_popup = None
        self.popup = None

    def build(self):
        return Builder.load_string(kv)

    def display_settings(self, settings):
        self.root.ids.box.add_widget(settings)
        return True

    def close_settings(self, *args):
        self.root.ids.box.clear_widgets()

    def on_start(self):
        Clock.schedule_interval(self._count, 1)

    def _count(self, _):
        self.count += 1
        print(f'Count: {self.count}')

    def open(self):
        self.open_settings()   # added to use as a breakpoint

    def pop(self):
        p = self.popup
        if not p:
            self.popup = Popup(content=Label(text='Pop'),
                                        title='Not Settings',
                                        size_hint=(0.8, 0.8))
            p = self.popup
        p.open()


CheckSettings().run()
```

**Comment by avonderlieth** [comment]
> Interesting the bug does not appear if kivy.uix.settings is imported prior to opening the settings panel: @avonderlieth this is a potential workaround to your issue.
> 
> I've extended the example to put the settings panel beside the Label so I could better see when the change was occurring.
> 

Hi Elliot,
thanks a lot - this works. Just some additional comment: it appears that the `import kivy.uix.settings` statement needs to come AFTER setting the graphics config "fullscreen" to "auto" or else this will not be set (at least in my example it didn't).

In case someone else is having issues with this, another workaround is to remove the widget and re-creating it after closing the settings (though @ElliotGarbus fix is much nicer).
Thanks again.

**Comment by misl6** [comment]
Fixed via #8535
